### PR TITLE
Service control with flags

### PR DESCRIPTION
**Pull request by troian**
start/stop/restart and check status by
cagent --service_start(stop|restart|status)
instead of running host specific commands

example

```
$ cagent --service_status
running
$ cagent --service_start
already
$ cagent --service_stop
stopped
$ cagent --service_start
started
```

### CODE REVIEW

File: cagent.go [hunk 1]
```diff
@@ -59,14 +59,15 @@ func New(cfg *Config, version string) *Cagent {
 
 	ca.SetLogLevel(ca.Config.LogLevel)
 
+	return ca
+}
+
+func (ca *Cagent) Initialize() error {
```
Comment (soupdiver) [inline cagent.go]: Is there a reason why we can't have `Initialize` `New`?

File: cmd/cagent/main.go [hunk 2]
```diff
@@ -105,16 +111,27 @@ func main() {
...
+	handleServiceCommand(ca, *flagServiceStatusPtr, *flagServiceStartPtr, *flagServiceStopPtr, *flagServiceRestartPtr)
+
+	handleFlagTest(*testConfigPtr, ca)
+
 	setDefaultLogFormatter()
 
+	// cagent Initialize must be called handleFlagPrintConfig and handleServiceCommand
+	// to prevent print logger setup error
+	if err = ca.Initialize(); err != nil {
+		if err != nil {
```
Comment (soupdiver) [inline cmd/cagent/main.go]: double checking for `if err != nil`

**Comment by troian** [comment]
Simplifies a bit interactions for plesk extension.
We can either write lot's of bash code to detect service manager type and how to control service in terms of start/stop/status.
Instead this PR adds few commands to control cagent service over kardianos/service package which already used by cagent.
Current PR looks a bit ugly but @requilence mentioned https://github.com/jessevdk/go-flags so resulting command can be `cagent service start`

**Comment by soupdiver** [comment]
> Simplifies a bit interactions for plesk extension.
> We can either write lot's of bash code to detect service manager type and how to control service in terms of start/stop/status.
> Instead this PR adds few commands to control cagent service over kardianos/service package which already used by cagent.
> Current PR looks a bit ugly but @requilence mentioned https://github.com/jessevdk/go-flags so resulting command can be `cagent service start`

ok, I see.

File: cagent.go (hunk 1 above)
Comment (troian) [reply cagent.go]: done

File: cmd/cagent/main.go [hunk 3]
```diff
@@ -105,7 +111,15 @@ func main() {
 
 	handleFlagPrintConfig(*printConfigPtr, cfg)
 
-	setDefaultLogFormatter()
+	if ((serviceInstallPtr == nil) || ((serviceInstallPtr != nil) && (!*serviceInstallPtr))) &&
```
Comment (troian) [inline cmd/cagent/main.go]: looks like https://github.com/jessevdk/go-flags suggests itself as this now look too ugly.
@soupdiver @requilence ?

Comment (requilence) [reply cmd/cagent/main.go]: agree. Now the CLI API becomes to look ugly 
image

Comment (soupdiver) [reply cmd/cagent/main.go]: I agree, it get's messy.
An opinion on grouping? E.g. `cagent service start` and maybe the other options could also be grouped... do we need backwards compatibility in this case?

File: cmd/cagent/main.go [hunk 4]
```diff
@@ -526,11 +604,18 @@ func getServiceFromFlags(ca *cagent.Cagent, configPath, userName string) (servic
...
 
-func setDefaultLogFormatter() {
+func setDefaultLogFormatter(ca *cagent.Cagent) {
 	tfmt := log.TextFormatter{FullTimestamp: true}
 	if runtime.GOOS == "windows" {
 		tfmt.DisableColors = true
 	}
 
 	log.SetFormatter(&tfmt)
+
```
Comment (requilence) [inline cmd/cagent/main.go]: the name of the function now doesn't show what is happening inside

Comment (troian) [reply cmd/cagent/main.go]: @requilence It still configures default log formatter. Isn't ?

Comment (requilence) [reply cmd/cagent/main.go]: from the other side, the function with name `setDefaultLogFormatter` now adds the file log hook which seems not quite expected

Comment (soupdiver) [reply cmd/cagent/main.go]: > now adds the file log hook which seems not quite expected

agree

Comment (troian) [reply cmd/cagent/main.go]: done

**Comment by soupdiver** [comment]
@requilence Can we get this merged?
Nearly three weeks old

**Pull request merged**